go.mod

```diff
@@ -43,7 +43,7 @@ require (
 	google.golang.org/grpc v1.59.0
 	k8s.io/api v0.30.1
 	k8s.io/apiextensions-apiserver v0.30.1
-	k8s.io/apimachinery v0.30.1
+	k8s.io/apimachinery v0.31.0-alpha.1
```

Comment:
question: Do we want to bump this to a different version? Could we still add `setup-envtest` with `v0.30.1` of this library?

Reply:
i didnt explicitly go get that version. it must've happened when i got setup-envtest. I can double check

Reply:
` go mod graph | grep k8s.io/apimachinery`
...
`sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240709080737-a0c9fd9d3f31 k8s.io/apimachinery@v0.31.0-alpha.1`